### PR TITLE
docs: release notes for v0.12.4

### PR DESCRIPTION
**Pull request by justin-determined-ai**
## Test Plan

`make`'d the docs locally, looks good!

## Commentary (optional)

Would like a quick sanity check against the release spreadsheet to ensure I'm not missing any commits we should add.

[DET-1234]: https://determinedai.atlassian.net/browse/DET-1234

### CODE REVIEW

File: docs/release-notes.txt [hunk 1]
```diff
@@ -6,6 +6,35 @@ Release Notes
...
 ------------
 
+Version 0.12.4
+^^^^^^^^^^^^^^
+
+**Release Date:** May 13, 2020
+
+- **Breaking Change:** Users are no longer automatically logged in as the "determined" user. Refer to the :ref:`users` documentation for more details.
+
+- Support multi-slot notebooks. The number of slots per notebook cannot exceed the size of largest available agent.
```
Comment (dzhu) [inline docs/release-notes.txt:16]: "the largest"

File: docs/release-notes.txt [hunk 2]
```diff
@@ -6,6 +6,35 @@ Release Notes
...
+Version 0.12.4
+^^^^^^^^^^^^^^
+
+**Release Date:** May 14, 2020
+
+- **Breaking Change:** Users are no longer automatically logged in as the "determined" user. Refer to the :ref:`users` documentation for more details.
+
+- Support multi-slot notebooks. The number of slots per notebook cannot exceed the size of largest available agent. 
+
+- Authentication sessions now expire every 7 days.
```
Comment (dzhu) [inline docs/release-notes.txt:18]: Perhaps "after"? It's not exactly a periodic event.

File: docs/release-notes.txt [hunk 3]
```diff
@@ -6,6 +6,35 @@ Release Notes
 Version 0.12
 ------------
 
+Version 0.12.4
+^^^^^^^^^^^^^^
+
+**Release Date:** May 14, 2020
+
+- **Breaking Change:** Users are no longer automatically logged in as the "determined" user. Refer to the :ref:`users` documentation for more details.
```
Comment (dzhu) [inline docs/release-notes.txt:14]:
```suggestion
- **Breaking Change:** Users are no longer automatically logged in as the "determined" user. Refer to :ref:`users` for more details.
```
or
```suggestion
- **Breaking Change:** Users are no longer automatically logged in as the "determined" user. Refer to the :ref:`users <users>` documentation for more details.
```

If there's no explicit title text given, I'm of the opinion that the reference should be treated as a location, without the requirement that the title of the page fit nicely into the sentence. Same for the other links.

File: docs/release-notes.txt (hunk 2 above)
Comment (justin-determined-ai) [reply docs/release-notes.txt:18]: lol true

File: docs/release-notes.txt (hunk 3 above)
Comment (justin-determined-ai) [reply docs/release-notes.txt:14]: Makes sense, good way of thinking about it.

**Pull request merged**